posts/2019/11/metadsl-talk.md

```diff
@@ -0,0 +1,150 @@
+
+It would be nice to see if we could convert between MLIR dialects and `metadsl`. And long term, if we could open a dialogue there and come together into one system in the end, that would be ideal!
+
+I have already started experimenting with translating `metadsl` types into JSON with the `typez` project. That's useful because then we could possibly generate these `typez` descriptions from MLIR dialects and then from there auto-generate Python code. Or vice versa.
+
+I think for now we should focus on doing something useful with `metadsl` first, to really stress test it, then we can come back and see how to bridge that gap.
+
+
+### Dataframes
+
+Which brings me to the most practically relevent conversations... Dataframes!
+
+I was excited to reconnect with Marc LAST NAME, after meeting him a few years ago at SciPy. Congratulations to him for winning a community (NAME OF AWARD) awards at the NumFocus summit last weekend! He ran a docs sprint for Pandas that drew XXX people and is mentoring XXX new women to contribute to Pandas.
+
+We talked about how we could think about targeting different backends and optimizing Pandas. His notion is that they are already exploring this for small chunks of computatoin. Like adding a `backend` flag to some larger operations to run them through different systems. They just implemented this for plotting as well. He said to move Pandas anywhere, it has to be incremental, that a full rewrite would be less likely to succeed.
+
+ 
+I was also thrilled to meet Jeff Reback for this first time, who heriocally helps manage the Pandas and Ibis projects, answering your issues and merging your pull requests. If you haven't hear of Ibis, it's a lazy Pandas project that can target different backends, like SQL databases. At Quansight, Ivan XXX has implemented a backend for the OmniSci GPU database (LINK TO BLOG POST OR WORK?), so that you can run queries over millions of rows very quickly from Python with a familiar API. This seems like a nice project to see if `metadsl` has legs.
+
+
+We should take a look at C# LINQ for inspiration here, to see how they had to update their type system to support it. We will have to also upstream some of [these changes][python-typing-dataframe] to Python's core typing.
+
+## Long term implications
+
+By describing our APIs with `metadsl` we transform our code into structured data. 
+
+### Reproducible Science
+
+Then, if we save this data, we are able to later analyze it semantically. For example, we can determine what data files were used to produce a certain visualization in a notebook if we ouput the computation graph alongside the result. This dovetails with [work we have been doing in JupyterLab to support viewing linked data][metadata-explorer] with [Paco Nathan][paco], [the NYU Coleridge Initiative][nyu], and CalPoly.
```

Comment:
I'm not sure how that would work, short of storing all inputs to all functions. How would you otherwise reconstruct some large amount of data - the same filename often points to changing contents over time.

Reply:
If you can have longer term pointers to data, than this helps. You need those anyway if you can even begin to have a hope of tracking semantics.